mantidproject/mantid
# Fix subplot alignment and legend placement in plotSubplots

``plotSubplots`` plots a multilayer graph with *n* subplots, each with a legend. This PR fixes several issues relating to the alignment of these plots and their legends. 

In addition, the plots now use MantidPlot's app-wide default plot settings, so they will look how the user expects.

**To test:**

The instructions for how to reproduce the three error cases are in the issue, or use the script below:
(Data available in `SampleData-ISIS.zip` from http://download.mantidproject.org/)
```python
mar11060 = Load('MAR11060')
mar11015 = Load('MAR11015')
mar11001 = Load('MAR11001')
gp = GroupWorkspaces([mar11060, mar11015, mar11001])

#test case 1 
plotSubplots(mar11060, [0,1,2])

#test case 2
plotSubplots(gp, [0,1,2])

#test case 3
g = plotSubplots(mar11060, [0,1])
plotSubplots(mar11060, [0,1], window=g)
```

In each case the legend placement and graph alignment should be fixed.

These are the results I see - compare to screenshots in the [issue comment](https://github.com/mantidproject/mantid/issues/17874#issuecomment-259990858):
**case 1**
![case1](https://cloud.githubusercontent.com/assets/15363125/20274027/37802852-aa8b-11e6-8a04-a8b50b5ae7df.PNG)

**case 2**
![case2](https://cloud.githubusercontent.com/assets/15363125/20274038/3c3f18a8-aa8b-11e6-8897-66fbd54b7e48.PNG)

**case 3**
![case3](https://cloud.githubusercontent.com/assets/15363125/20274043/40a2ea32-aa8b-11e6-9d60-3d6adb52dcfb.PNG)


Fixes #17874.

[Release notes](https://github.com/mantidproject/mantid/blob/e2a17bd5c7b4b3c5ad0085044464bf2dd6f610db/docs/source/release/v3.9.0/ui.rst#bugs-resolved) have been updated.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
